exchanges/okcoin/okcoin_websocket.go

```diff
@@ -147,150 +238,268 @@ func (o *OKCoin) WsHandleData(respRaw []byte) error {
 	var eventResponse WebsocketEventResponse
 	err = json.Unmarshal(respRaw, &eventResponse)
 	if err == nil && eventResponse.Event != "" {
-		if eventResponse.Event == "login" {
-			if o.Websocket.Match.Incoming("login") {
-				o.Websocket.SetCanUseAuthenticatedEndpoints(eventResponse.Success)
+		switch eventResponse.Event {
+		case "login":
+			if o.Websocket.Match.IncomingWithData("login", respRaw) {
+				o.Websocket.SetCanUseAuthenticatedEndpoints(eventResponse.Code == "0")
+			}
+		case "subscribe", "unsubscribe":
+			o.Websocket.DataHandler <- eventResponse
+		case "error":
```

Comment:
With `waitingSignatures` I'm not seeing any possibility of the event type being "error":
https://www.okcoin.com/docs-v5/en/#websocket-api-trade-cancel-order

Reply:
Embarrassingly, The error event I tried to handle is found to be for "subscriptions", not for the WebSocket "trade". The subscriptions WebSocket class doesn't use their signature, therefore, I removed the `waitingSignatureLock, waitingSignature` implementation.